keras_cv/layers/preprocessing/cut_mix.py

```diff
@@ -137,10 +166,10 @@ def _fill_rectangle(
         tf.zeros(cutout_shape, dtype=image.dtype), padding_dims, constant_values=1
     )
     mask = tf.expand_dims(mask, -1)
-    mask = tf.tile(mask, [1, 1, 3])
+    mask = tf.tile(mask, [1, 1, image_shape[-1]])
```

Comment:
Make sure to test it in graph/tf.function model as well, since tf.shape will return a tensor in this case, concatng ints with tensor might raise some error.

Reply:
Fixed